### PR TITLE
CRS comparison performance improvements

### PR DESCRIPTION
**Pull request by metasim**
To address https://github.com/locationtech/rasterframes/issues/134

Note: Merge https://github.com/locationtech/rasterframes/pull/132 first.

### CODE REVIEW

File: core/src/main/scala/org/locationtech/rasterframes/expressions/transformers/ReprojectGeometry.scala [hunk 1]
```diff
@@ -34,15 +34,26 @@ import org.apache.spark.sql.types.DataType
...
+    * dstCRS - the CRS to project geometry into""",
+  examples = """
+  Examples:
+    > SELECT _FUNC_(geom, srcCRS, dstCRS);
+       ..."""
+)
 case class ReprojectGeometry(geometry: Expression, srcCRS: Expression, dstCRS: Expression) extends Expression
   with CodegenFallback with ExpectsInputTypes {
 
+  // TODO: Replace registration in `org.locationtech.rasterframes.functions.register`
```
Comment (vpipkt) [inline core/src/main/scala/org/locationtech/rasterframes/expressions/transformers/ReprojectGeometry.scala]: If it makes sense, let's hold off on merging this PR until we handle this TODO

File: core/src/main/scala/org/locationtech/rasterframes/expressions/transformers/ReprojectGeometry.scala [hunk 2]
```diff
@@ -60,10 +71,15 @@ case class ReprojectGeometry(geometry: Expression, srcCRS: Expression, dstCRS: E
     }
 
   override def eval(input: InternalRow): Any = {
-    val geom = JTSTypes.GeometryTypeInstance.deserialize(geometry.eval(input))
     val src = srcCRS.eval(input).asInstanceOf[InternalRow].to[CRS]
     val dst = dstCRS.eval(input).asInstanceOf[InternalRow].to[CRS]
-    JTSTypes.GeometryTypeInstance.serialize(reproject(geom, src, dst))
+    (src, dst) match {
+      // Optimized pass-through case.
+      case (s: LazyCRS, r: LazyCRS) if s.encoded == r.encoded => geometry.eval(input)
```
Comment (vpipkt) [inline core/src/main/scala/org/locationtech/rasterframes/expressions/transformers/ReprojectGeometry.scala]: 👍

File: core/src/main/scala/org/locationtech/rasterframes/functions/package.scala [hunk 3]
```diff
@@ -149,8 +149,8 @@ package object functions {
 
     sqlContext.udf.register("rf_cell_types", cellTypes)
     sqlContext.udf.register("rf_rasterize", rasterize)
+    // TODO: replace this with full ReprojectGeometry with String handling.
```
Comment (vpipkt) [inline core/src/main/scala/org/locationtech/rasterframes/functions/package.scala]: as above let's get all the function registration cleaned up before merge.

Comment (metasim) [reply core/src/main/scala/org/locationtech/rasterframes/functions/package.scala]: I feel like that's a separate PR.... trying to move away from my habit of monolithic PRs that take forever to review!

File: core/src/main/scala/org/locationtech/rasterframes/expressions/transformers/ReprojectGeometry.scala (hunk 1 above)
Comment (metasim) [reply core/src/main/scala/org/locationtech/rasterframes/expressions/transformers/ReprojectGeometry.scala]: https://github.com/locationtech/rasterframes/issues/138

**Pull request merged**